EasyCommands/BlockHandlers/TurretBlockHandlers.cs

```diff
@@ -0,0 +1,87 @@
+                    TypeHandler(VectorHandler(b => {
+                        Vector3D? target = Vector3D.Zero;
+                        if (b.HasTarget) target = GetPosition(b.GetTargetedEntity());
+                        return target.Value;
+                    }), Return.VECTOR),
+                    TypeHandler(BooleanHandler(b => b.HasTarget), Return.BOOLEAN),
+                    TypeHandler(StringHandler(b => b.GetTargetingGroup(), (b, v) => b.SetTargetingGroup(v)), Return.STRING));
+                AddPropertyHandler(Property.ANGLE, TerminalPropertyHandler("AngleDeviation", 5f));
+                AddBooleanHandler(Property.TRIGGER, b => GetGuns(b).Any(g => g.IsShooting), (b, v) => {
+                    foreach (IMyUserControllableGun gun in GetGuns(b)) gun.Shoot = v;
+                });
+                AddDirectionHandlers(Property.VELOCITY, Direction.RIGHT,
+                    TypeHandler(NumericHandler(b => b.VelocityMultiplierAzimuthRpm, (b,v) => b.VelocityMultiplierAzimuthRpm = v), Direction.LEFT, Direction.RIGHT),
+                    TypeHandler(NumericHandler(b => b.VelocityMultiplierElevationRpm, (b,v) => b.VelocityMultiplierElevationRpm = v), Direction.UP, Direction.DOWN));
+                defaultPropertiesByPrimitive[Return.NUMERIC] = Property.RANGE;
+            }
+
+            IEnumerable<IMyUserControllableGun> GetGuns(IMyTurretControlBlock block) {
+                var tools = NewList<IMyFunctionalBlock>();
+                block.GetTools(tools);
+                return tools.Where(t => t is IMyUserControllableGun).Select(t => (IMyUserControllableGun)t);
+            }
+        }
+
+        public class TurretBlockHandler<T> : GunBlockHandler<IMyLargeTurretBase> {
+            public TurretBlockHandler() {
+                AddNumericHandler(Property.RANGE, b => b.Range, (b,v) => b.Range = v, 100f);
+                AddPropertyHandler(Property.LOCKED, TerminalPropertyHandler("EnableTargetLocking", true));
+                AddBooleanHandler(Property.USE, b => b.IsUnderControl);
+                AddBooleanHandler(Property.ROLL_INPUT, b => b.EnableIdleRotation, (b,v) => b.EnableIdleRotation = v);
+                AddReturnHandlers(Property.TARGET, Return.VECTOR,
+                    TypeHandler(VectorHandler(GetTarget, SetTarget), Return.VECTOR),
+                    TypeHandler(BooleanHandler(b => b.HasTarget, (b, v) => { if (!v) ResetTarget(b); }), Return.BOOLEAN),
+                    TypeHandler(StringHandler(b => b.GetTargetingGroup(), (b, v) => b.SetTargetingGroup(v)), Return.STRING));
+                AddVectorHandler(Property.TARGET_VELOCITY, b => b.GetTargetedEntity().Velocity, (b, v) => b.TrackTarget(GetTarget(b), v));
+                AddNumericHandler(Property.ANGLE, b => b.Azimuth * RadiansToDegrees, (b, v) => b.Azimuth = v * DegreesToRadians, 5);
+                AddNumericHandler(Property.ALTITUDE, b => b.Elevation * RadiansToDegrees, (b, v) => b.Elevation = v * DegreesToRadians, 5);
```

Comment:
azimuth and elevation needs to synchronized (`SyncAzimuth()`, `SyncElevation()`) if i read the mdk wiki right.